### ISSUE TITLE
"View in context" links goes to empty conversation page

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#12059]
See https://friendica.mrpetovan.com/search?q=programmers
Original post: https://pirati.ca/display/ec054ce7-3563-48ee-deaf-98b590748752
Local page: https://friendica.mrpetovan.com/display/ec054ce7-3563-48ee-deaf-98b590748752

What should I test to figure this one out?

**Comment by tobiasd** [comment]
Could be a problem with being logged in or not? For me your links to the original post and the local page show (almost as the user avatar is missing in one) the same thread. Also following the context link gives me the correct thread.

However, looking at the page locally on my node, I get an page not found error caused by the posting not existing on my node.

**Comment by MrPetovan** [comment]
Oooh, the blank page (but with the correct author aside?) shows when I'm logged in.

**Comment by MrPetovan** [comment]
I found the issue: I currently ignore the author of the post shared in the top-level post of this conversation. The top-level post isn't fetched, and the conversation isn't built because the starting item is the comment referenced by the GUID, not a top-level post. No top-level post, no conversation.

I'm going to need a more permissive query on `post-user-view` for this module because it's about a single conversation where the top-level post should show no matter what (blocks, ignores, etc...)

**Comment by MrPetovan** [comment]
I ended up giving up on showing the conversation because the `Display` module uses the same underlying code in `Conversation` than the `Network` page that very much needs to enforce user moderation decisions.

Instead I opted to display a detailed error page with most reasons a top-level post wouldn't show for the current user. I based myself on the condition in `Model\Post::selectViewForUser` but I may have missed something.
